google/cloud/aiplatform/training_jobs.py

```diff
@@ -281,46 +333,474 @@ def package_and_copy_to_gcs(
+            worker_pool_spec["machineSpec"]["acceleratorType"] = accelerator_type
+            worker_pool_spec["machineSpec"]["acceleratorCount"] = accelerator_count
+
+        if args:
+            worker_pool_spec["pythonPackageSpec"]["args"] = format_args(args)
+
+        managed_model = None
+        # create model payload
+        if model_display_name:
+
+            container_spec = ModelContainerSpec(
+                image_uri=self._model_serving_container_image_uri,
+                predict_route=self._model_serving_container_predict_route,
+                health_route=self._model_serving_container_health_route,
+            )
+
+            managed_model = Model(
+                display_name=model_display_name, container_spec=container_spec
+            )
+
+        input_data_config = None
+        if dataset:
+            # Create fraction split spec
+            fraction_split = FractionSplit(
+                training_fraction=training_fraction_split,
+                validation_fraction=validation_fraction_split,
+                test_fraction=test_fraction_split,
+            )
+
+            # create input data config
+            input_data_config = InputDataConfig(
+                fraction_split=fraction_split,
+                dataset_id=dataset.name,
+                gcs_destination=GcsDestination(output_uri_prefix=base_output_dir),
+            )
+
+        # create training pipeline
+        training_pipeline = TrainingPipeline(
+            display_name=self._display_name,
+            training_task_definition=schema.training_job.definition.custom_task,
+            training_task_inputs=json_format.ParseDict(
+                {
+                    "workerPoolSpecs": [worker_pool_spec],
+                    "baseOutputDirectory": {"output_uri_prefix": base_output_dir},
+                },
+                Value(),
+            ),
+            model_to_upload=managed_model,
+            input_data_config=input_data_config,
+        )
+
+        training_pipeline = self.api_client.create_training_pipeline(
+            parent=initializer.global_config.common_location_path(
+                self.project, self.location
+            ),
+            training_pipeline=training_pipeline,
+        )
+
+        self._gca_resource = training_pipeline
+
+        _LOGGER.info("View Training:\n%s" % self._dashboard_uri())
+        _LOGGER.info("Training Output directory:\n%s " % base_output_dir)
+
+        model = self._get_model()
+
+        if model is None:
+            _LOGGER.warn(
+                "Training did not produce a Managed Model returning None. "
+                + self._model_upload_fail_string
+            )
+        return model
+
+    def _sync_gca_resource(self):
+        """Helper method to sync the local gca_source against the service."""
+        self._gca_resource = self.api_client.get_training_pipeline(
+            name=self.resource_name
+        )
+
+    def _block_until_complete(self):
```

Comment:
for later: This (and maybe other) helper methods might be abstracted to deal with other kinds of `Job`s e.g. batch prediction job.

Reply:
Agree.